libphobos/libdruntime/core/cpuid.d

```diff
@@ -934,6 +970,42 @@ bool hasCPUID()
     }
 }
 
+version (GNU)
+{
+    nothrow @nogc
+    void gnuCpuid(uint eax, out uint a, out uint b, out uint c, out uint d)
+    {
+        gnuCpuid(eax, 0, a, b, c, d);
+    }
+
+    nothrow @nogc
+    void gnuCpuid(uint eax, uint ecx, out uint a, out uint b, out uint c, out uint d)
+    {
+        // On x86 the PIC register 'ebx' must not be overwritten. GCC <5 will error out otherwise.
+        version (X86) version (D_PIC)
+            enum SaveEbx;
+        
+        static if (is(SaveEbx))
+        {
+            asm nothrow @nogc
+            {
+                "push {%%}ebx; cpuid; mov {%%ebx, %1|%1, ebx}; pop {%%}ebx"
+                    : "=a" a, "=r" b, "=c" c, "=d" d            // output
```

Comment:
You mean 'ebx', not 'rbx', right? I understood where it is "early-clobbered" now. The compiler could otherwise choose to use 'eax' or 'ecx' for it, before those are consumed by the 'cpuid'. I'll update the PR.